app/controllers/deploys_controller.rb

```diff
@@ -1,18 +1,13 @@
 class DeploysController < ApplicationController
-  include CurrentProject
   include ProjectLevelAuthorization
 
-
-  before_action except: [:active, :active_count, :recent, :changeset] do
-    find_project(params[:project_id])
-  end
-
```

Comment:
Rails is an opinionated framework. It does a whole lot of magic for you behind the scenes, and sacrifices explicitness for ease of use, and needing minimal boilerplate code. When I'm thinking of inserting some "magic" code like this, I ask myself "does this follow patterns that Rails developers are familiar with, or would this be surprising to them?"

The pattern of loading certain required records as part of controllers is pretty standard. For example, you don't have to call a method to define `current_user` in every action of every controller. That's just done for you as part of authorization.

Similarly, a large number of our controller actions don't make sense unless you have a `current_project`.  If you tried going to http://samson.example.com/projects/NOT_A_PROJECT/deploys, it should immediately error or redirect you without ever running the action. So given that's a necessary check, and since it shows up in so many controllers, it makes sense to me to have it in a `before_filter`. Actually, maybe call the filter something like `require_project` to make it more explicit that it's doing a check, and not just loading something from the DB.

So in this case, in my opinion, abstracting out the loading and checking of `current_project` into a module and having that happen automatically is worth the slight loss of explicitness.  But it's obviously not something with a clear-cut answer to it.